decoder.go

```diff
@@ -90,10 +95,10 @@ func (d *Decoder) DecodeTag() (tag int, wireType WireType, err error) {
 	}
 	v, n, err := DecodeVarint(d.p[d.offset:])
 	if err != nil {
-		return 0, -1, err
+		return 0, -1, fmt.Errorf("invalid data at byte %d: %w", d.offset, err)
 	}
-	if n < 1 || n > MaxTagValue {
-		return 0, -1, ErrInvalidFieldTag
+	if n < 1 || v < 1 || v > MaxTagValue {
```

Comment:
Should you add an explicit case for where the tag is 0 in your test at decoder_test.go::FuzzDecodeTag?

Reply:
i.e. add `{(0 << 3)},     // tag=0, wire type=0`
I noticed that if I remove the `case errors.Is(err, csproto.ErrInvalidFieldTag)` case and add some debug logging to DecodeTag(), it never fails because v == 0. Might be nice to force that condition.

Reply:
Interestingly, I generated that failure almost immediately locally when I ran the fuzz testing and had to add the new case to that `switch`.

Reply:
Ya, same... but it was always with a non-zero v value... specifically it was throwing the error because v was some value > than MaxTagValue